lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -993,7 +993,7 @@ static unsigned IsPtrUsedByLoweredFn(
 
     } else if (ConstantExpr *CE = dyn_cast<ConstantExpr>(user)) {
       unsigned opcode = CE->getOpcode();
-      if (opcode == Instruction::AddrSpaceCast || Instruction::GetElementPtr)
+      if (opcode == Instruction::AddrSpaceCast || opcode == Instruction::GetElementPtr)
```

Comment:
This is a completely unrelated change that I noticed and added incidentally since the previous conditional was always true, I'm certain it wasn't what was intended.

Reply:
Good catch!